pkg/provisioner/ironic/findhost_test.go

```diff
@@ -0,0 +1,77 @@
+package ironic
+
+import (
+	"testing"
+
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/clients"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/testserver"
+)
+
+func TestFindExistingHost(t *testing.T) {
+	cases := []struct {
+		name   string
+		ironic *testserver.MockServer
+
+		hostName       string
+		provisioningID string
+		nodeName       string
+	}{
+		{
+			name:           "no-node",
+			hostName:       "name",
+			provisioningID: "uuid",
+			ironic: testserver.New(t, "ironic").NotFound("/v1/nodes/name").
+				NotFound("/v1/nodes/uuid"),
+		},
+		{
+			name:           "by-name",
+			hostName:       "name",
+			provisioningID: "uuid",
+			ironic: testserver.New(t, "ironic").NotFound("/v1/nodes/uuid").
+				Response("/v1/nodes/name", `{"name": "name", "uuid": "different-uuid"}`),
+			nodeName: "name",
+		},
+		{
+			name:           "by-uuid",
+			hostName:       "name",
+			provisioningID: "uuid",
+			ironic: testserver.New(t, "ironic").NotFound("/v1/nodes/name").
+				Response("/v1/nodes/uuid", `{"name": "different-name", "uuid": "uuid"}`),
+			nodeName: "different-name",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.ironic != nil {
+				tc.ironic.Start()
+				defer tc.ironic.Stop()
+			}
+
+			auth := clients.AuthConfig{Type: clients.NoAuth}
+
+			// Update the default host to match the test settings
+			host := makeHost()
+			host.ObjectMeta.Name = tc.hostName
+			host.Status.Provisioning.ID = tc.provisioningID
+
+			prov, err := newProvisionerWithSettings(host, bmc.Credentials{}, nil,
+				tc.ironic.Endpoint(), auth, "https://inspector.test/", auth,
+			)
```

Comment:
A minor point, I noticed that this setup flow (create the auths, build an host - here with some customizations) and the call `newProvisionerWithSettings` seems to be pretty common, maybe we could have some test scenario builder dedicated for that to remove this duplication

Reply:
Yes, good point. I noticed that I was doing a lot of copying and pasting as I came up with the tests, but I was focused on the mock API server and not so much at this level.

I expect to keep working on these tests in future PRs, so I will look at ways to consolidate that further as I add more tests.